### PR TITLE
Tag emails with `message_type` in headers

### PR DESCRIPTION
**Pull request by benbarnett**
Allows for more granular analysis of email deliveries and
interaction.

### CODE REVIEW

File: app/mailers/application_mailer.rb [hunk 1]
```diff
@@ -1,8 +1,7 @@
+require 'mailgun_utils'
+
 class ApplicationMailer < ActionMailer::Base
-  default(
-    from: 'appointments@pensionwise.gov.uk',
-    'X-Mailgun-Variables' => { online_booking: true }.to_json
-  )
+  default(from: 'appointments@pensionwise.gov.uk')
```
Comment (benlovell) [inline app/mailers/application_mailer.rb]: Can we remove the parens from this now?

File: app/mailers/appointments.rb [hunk 2]
```diff
@@ -1,10 +1,13 @@
 class Appointments < ApplicationMailer
+  include MailgunUtils
```
Comment (benlovell) [inline app/mailers/appointments.rb]: May as well just include this in `ApplicationMailer` once instead?

File: lib/mailgun_utils.rb [hunk 3]
```diff
@@ -0,0 +1,14 @@
+module MailgunUtils
```
Comment (benlovell) [inline lib/mailgun_utils.rb]: Nit-picky but the name feels a bit scuzzy. Can we call this `MailgunHeaders` or something instead?

Comment (benbarnett) [inline lib/mailgun_utils.rb]: Yep, good idea.

**Comment by benlovell** [comment]
Lovely. 🚢 that 💩

**Pull request merged**